catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java

```diff
@@ -288,22 +265,76 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
 	@Override
 	public void onPanelSlide(View panel, float slideOffset) {
-		converterSlidingUpPanelFragment.rotateImageButton(slideOffset * 180.0f);
+		getSlidingUpFragment().rotateImageButton(slideOffset * 180.0f);
 	}
 
 	@Override
 	public void onPanelStateChanged(View panel, SlidingUpPanelLayout.PanelState previousState,
 			SlidingUpPanelLayout.PanelState newState) {
-		Log.d(TAG, "SlidingUpPanel state changed: " + newState.toString());
 		switch (newState) {
 			case EXPANDED:
-				converterSlidingUpPanelFragment.rotateImageButton(180);
-				converterSlidingUpPanelFragment.scrollUpPanelScrollView();
+				getSlidingUpFragment().rotateImageButton(180);
+				getSlidingUpFragment().scrollUpPanelScrollView();
 				break;
 			case COLLAPSED:
-				converterSlidingUpPanelFragment.rotateImageButton(0);
-				converterSlidingUpPanelFragment.scrollUpPanelScrollView();
+				getSlidingUpFragment().rotateImageButton(0);
+				getSlidingUpFragment().scrollUpPanelScrollView();
 				break;
 		}
 	}
+
+	public void onItemClick(Job job) {
+		if (!Looper.getMainLooper().equals(Looper.myLooper())) {
+			throw new AssertionError("You should not change the UI from any thread except UI thread!");
+		}
+
+		if (job == null) {
+			Log.e(TAG, "Job not found in runningJobsAdapter!");
+			return;
+		}
+
+		if (job.getState() == Job.State.FAILED) {
+			ToastUtil.showError(this, R.string.error_cannot_open_failed_scratch_program);
+			return;
+		}
+
+		String catrobatProgramName = getSlidingUpFragment().getDownloadedProgramsMap().get(job.getJobID());
+		catrobatProgramName = catrobatProgramName == null ? job.getTitle() : catrobatProgramName;
+
+		if (job.getDownloadState() == Job.DownloadState.DOWNLOADING) {
+			AlertDialog.Builder builder = new CustomAlertDialogBuilder(this);
```

Comment:
Don't use the CustomAlertDialogBuilder. I should have told you before, but it is going to be removed (with #2729), please just change it to AlertDialog.Builder.